tests/0112-assign_unknown_part.c

```diff
@@ -68,17 +66,8 @@ int main_0112_assign_unknown_part(int argc, char **argv) {
         TEST_SAY("Changing partition count for topic %s\n", topic);
         test_create_partitions(NULL, topic, 2);
 
-        /* FIXME: The new partition might not have propagated through the
-         *        cluster by the time the producer tries to produce to it
-         *        which causes the produce to fail.
-         *        Loop until the partition count is correct. */
-        while ((r = test_get_partition_count(c, topic, 5000)) != 2) {
-                TEST_SAY(
-                    "Waiting for %s partition count to reach 2, "
-                    "currently %d\n",
-                    topic, r);
-                rd_sleep(1);
-        }
+        /* Allow the partition to propagate */
+        rd_sleep(1);
```

Comment:
This is for the same reason that KRaft replies more quickly and we need to await metadata propagation.
I can create a `test_create_partitions_wait_exist` in case there will be a quicker way to do that in future.